client/my-sites/promote-post-i2/hooks/use-open-promote-widget.ts

```diff
@@ -17,14 +17,19 @@ export interface Props {
 const useOpenPromoteWidget = ( { keyValue, entrypoint, external }: Props ) => {
 	const { openModal } = useRouteModal( 'blazepress-widget', keyValue );
 	const siteId = useSelector( getSelectedSiteId );
+	const isRunningInJetpack = config.isEnabled( 'is_running_in_jetpack_site' );
+	const isRunningInClassicSimple = useSelector( ( state ) =>
+		getSiteOption( state, siteId, 'is_wpcom_simple' )
+	);
+	const isRunningJetpackOrClassicSimple = isRunningInJetpack || isRunningInClassicSimple;
```

Comment:
Nitpick (non-blocking): We have this check now duplicated in a few places. I wonder if we should centralize in a common function that's reused everywhere so it's easier to maintain future changes to this logic.

Reply:
Yup, let's do that in this followup issue: https://github.com/Automattic/dotcom-forge/issues/6569

Reply:
> Something unrelated to this PR that I noted though is that, if I quit the promote process I'm not redirected back to where I was before:

I'll confirm to see if this is the same behavior for self-hosted/atomic sites, if it is, I feel like we can work on keeping behavior consistent and revisit this as a whole together.